extensions/amp-image-viewer/0.1/amp-image-viewer.js

```diff
@@ -119,61 +109,85 @@ export class ImageViewer {
     this.maxX_ = 0;
     /** @private {number} */
     this.maxY_ = 0;
-    /** @private {?./gesture.Gestures} */
+
+    /** @private {?../../../src/gesture.Gestures} */
     this.gestures_ = null;
-    /** @private {?function()} */
-    this.unlistenOnSwipePan_ = null;
 
-    /** @private {?./motion.Motion} */
+    /** @private {?../../../src/motion.Motion} */
     this.motion_ = null;
 
-    this.setupGestures_();
+    /** @private {?Element} */
+    this.sourceElement_ = null;
   }
 
-  /**
-   * Returns the root element of the image viewer.
-   * @return {!Element}
-   */
-  getElement() {
-    return this.viewer_;
+  /** @override */
+  buildCallback() {
+    this.vsync_ = this.getVsync();
+    this.element.classList.add('i-amphtml-image-lightbox-viewer');
+    const children = this.getRealChildren();
+    dev().assert(children.length == 1);
```

Comment:
given we may actually release this as a component, `user().assert` would be better.
I thin we should asserts that children.length == 1 and children[0].tagName == 'AMP-IMG' and have a message saying `One and only one `amp-img` must be a child of `amp-image-viewer` if it fails.

Reply:
Currently speaking, it can also be an `img`. Should we also allow the `img` tag, or is it okay to be so strict?

Reply:
when publically exposing this as an AMP component, `img` won't be allowed. Must be `amp-img`

Reply:
I see. Got it.